File: hybrid_inheritance.py
# Hybrid Inheritance is a combination of two or more types of inheritance.

class Animal:
    def eat(self):
        print("Eating")


class Mammal(Animal):
    def feed_milk(self):
        print("Feeding milk")


class Bird(Animal):
    def lay_eggs(self):
        print("Laying eggs")


class Bat(Mammal, Bird):
    pass


bat = Bat()
bat.feed_milk()
bat.lay_eggs()
bat.eat()

# In the above example, the Bat class inherits from both the Mammal and Bird classes. This is an example of Hybrid Inheritance. The Bat class has access to the eat method from the Animal class, as well as the feed_milk method from the Mammal class and the layEggs method from the Bird class.

File: hybrid_inheritance.cpp
// Hybrid Inheritance is a combination of two or more types of inheritance.

#include <bits/stdc++.h>
using namespace std;

class Animal
{
public:
    void eat()
    {
        cout << "Eating" << endl;
    }
};

class Mammal : public Animal
{
public:
    void feedMilk()
    {
        cout << "Feeding milk" << endl;
    }
};

class Bird : public Animal
{
public:
    void layEggs()
    {
        cout << "Laying eggs" << endl;
    }
};

class Bat : public Mammal, public Bird
{
};

int main()
{
    Bat obj;
    obj.feedMilk();
    obj.layEggs();
    obj.eat();
    return 0;
}

// In the above code, we have a base class Animal, two derived classes Mammal and Bird, and a class Bat that inherits from both Mammal and Bird classes. The Mammal class has a method feedMilk() and the Bird class has a method layEggs(). The Bat class inherits both the feedMilk() method from the Mammal class and the layEggs() method from the Bird class. When we create an object of the Bat class, it can access methods from both the Mammal and Bird classes. This is an example of hybrid inheritance.